### PR TITLE
Fix fn_listextendedproperty() to properly determine collation after ANALYZE

### PR DESCRIPTION
**Pull request by Sairakan**
The definition of the pl/pgsql sys.fn_listextendedproperty() function used
implicit collations in several places, which was fixed by adding explicit
COLLATE "C" clauses in a few places. Since this function is defined in the
postgres dialect but executed from the tsql dialect, where the default
collation is different, it is sometimes necessary to add explicit collations
when comparing string expressions.

Signed-off-by: Jason Teng <jasonten@amazon.com>

### Description

[Describe what this change achieves]
 
### Issues Resolved

JIRA issue BABEL-3279

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql [hunk 1]
```diff
@@ -2957,6 +2957,49 @@ SELECT
 WHERE FALSE; -- This condition will ensure that the view is empty
 GRANT SELECT ON sys.numbered_procedures TO PUBLIC;
 
+CREATE OR REPLACE FUNCTION sys.fn_listextendedproperty (
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql:3098]: please add upgrade tests to validate calling this function from a view/function/procedure across release works

Comment (Sairakan) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql:3098]: Added the tests

**Review by 2jungkook — DISMISSED** [review]

File: test/JDBC/upgrade/13_5/schedule [hunk 2]
```diff
@@ -75,3 +75,4 @@ triggers_with_transaction
 Test-sp_helpsrvrolemember
```
Comment (surendravishnoi) [inline test/JDBC/upgrade/13_5/schedule]: please add to 13_4 schedule as well if applicable

**Review by surendravishnoi — DISMISSED** [review]

**Review by JimFinnerty — DISMISSED** [review]

**Review by 2jungkook — DISMISSED** [review]

**Comment by Sairakan** [comment]
Currently one of the upgrade tests is failing on the BABEL-2701 test, but I think this is a flaky test that needs to be rewritten/removed from the test runs. See https://github.com/babelfish-for-postgresql/babelfish_extensions/commit/5818c532b533ba2d21e287049f3a6f8d01cebfc6 for another commit that was merged with this test failing in the checks as well.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 3]
```diff
@@ -1673,11 +1673,11 @@ value	sys.sql_variant
 as $$
 begin
 -- currently only support COLUMN property
-IF (((SELECT coalesce(property_name, '')) = '') or
-    ((SELECT coalesce(property_name, '')) = 'COLUMN')) THEN
-	IF (((SELECT coalesce(level0_object_type, '')) = 'schema') and
-	    ((SELECT coalesce(level1_object_type, '')) = 'table') and
-	    ((SELECT coalesce(level2_object_type, '')) = 'column')) THEN
+IF (((SELECT coalesce(property_name COLLATE "C", '')) = '') or
```
Comment (nasbyj) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: None of these SELECTs are needed; execution is implied by the IF.

**Review by priyansx — DISMISSED** [review]
LGTM

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 4]
```diff
@@ -1686,8 +1686,8 @@ IF (((SELECT coalesce(property_name, '')) = '') or
 		from sys.extended_properties t1, pg_catalog.pg_class t2, information_schema.columns t3
 		where t1.major_id = t2.oid and 
 			  t2.relname = t3.table_name and 
-		      t2.relname = (SELECT coalesce(level1_object_name, '')) and 
-			  t3.column_name = (SELECT coalesce(level2_object_name, ''));
+              t2.relname = (SELECT coalesce(level1_object_name COLLATE "C", '')) and
```
Comment (nasbyj) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql]: Same, no need for SELECT.

**Review by nasbyj — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 3 above)
Comment (Sairakan) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Removed

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 4 above)
Comment (Sairakan) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql]: Removed

**Review by priyansx — APPROVED** [review]
LGTM

**Review by 2jungkook — APPROVED** [review]

**Review by nasbyj — APPROVED** [review]

**Pull request merged**